gui-agent/vmside.c

```diff
@@ -898,10 +901,14 @@ static void process_xevent_map(Ghandles * g, XID window)
 static void process_xevent_unmap(Ghandles * g, XID window)
 {
     struct msg_hdr hdr;
+    struct window_data *wd;
     SKIP_NONMANAGED_WINDOW;
 
+    wd = list_lookup(windows_list, window)->data;
```

Comment:
Since you do list_lookup here anyway, you can open-code (modified) `SKIP_NONMANAGED_WINDOW` here to avoid traversing the list twice.